Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryBuilder.java

```diff
@@ -2344,6 +2344,114 @@ public Expression resolveIdentifier(String identifier, boolean mustResolve) {
         return null;
     }
 
+    public QueryLetRef getQueryLetRef(LetClause let) {
+        QueryLetRef result = of.createQueryLetRef().withName(let.getIdentifier());
+        result.setResultType(let.getResultType());
+        return result;
+    }
+
+    public ValueSetRef getValueSetRef(ValueSetDef valueSetDef) {
+        checkLiteralContext();
+        ValueSetRef valuesetRef = of.createValueSetRef().withName(valueSetDef.getName());
+        valuesetRef.setResultType(valueSetDef.getResultType());
+        if (valuesetRef.getResultType() == null) {
+            // ERROR:
+            throw new IllegalArgumentException(String.format("Could not validate reference to valueset %s because its definition contains errors.",
+                    valuesetRef.getName()));
+        }
+        if (isCompatibleWith("1.5")) {
+            valuesetRef.setPreserve(true);
+        }
+
+        return valuesetRef;
+    }
+
+    public Expression getCodeRef(CodeDef codeDef) {
+        checkLiteralContext();
+        CodeRef codeRef = of.createCodeRef().withName((codeDef).getName());
+        codeRef.setResultType(codeDef.getResultType());
+        if (codeRef.getResultType() == null) {
+            // ERROR:
+            throw new IllegalArgumentException(String.format("Could not validate reference to code %s because its definition contains errors.",
+                    codeRef.getName()));
+        }
+        return codeRef;
+    }
+
+    public Expression getCodeSystemRef(CodeSystemDef codeSystemDef) {
+        checkLiteralContext();
+        CodeSystemRef codesystemRef = of.createCodeSystemRef().withName(codeSystemDef.getName());
+        codesystemRef.setResultType(codeSystemDef.getResultType());
+        if (codesystemRef.getResultType() == null) {
+            // ERROR:
+            throw new IllegalArgumentException(String.format("Could not validate reference to codesystem %s because its definition contains errors.",
+                    codesystemRef.getName()));
+        }
+        return null;
+    }
+
+    public ParameterRef resolveParameterRef(ParameterDef theParameterDef) {
+        final ParameterRef parameterRef = of.createParameterRef().withName(theParameterDef.getName());
+        parameterRef.setResultType(parameterRef.getResultType());
+        return parameterRef;
+    }
+
+    public OperandRef resolveOperandRef(OperandDef operandDef) {
```

Comment:
suggestion: same as above resolve -> create

Reply:
I deleted this method since I no longer need to deal with Expressions.